core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -185,25 +185,34 @@ public CoreEvent process(CoreEvent event) throws MuleException {
   private BiConsumer<Throwable, Object> getContinueStrategyErrorHandler(Processor processor) {
     return (throwable, object) -> {
       if (object != null && !(object instanceof CoreEvent)) {
-        LOGGER.error(UNEXPECTED_ERROR_HANDLER_STATE_MESSAGE, throwable);
-        throw new IllegalStateException(UNEXPECTED_ERROR_HANDLER_STATE_MESSAGE);
-      }
-      CoreEvent event = (CoreEvent) object;
-      throwable = Exceptions.unwrap(throwable);
-      if (throwable instanceof MessagingException) {
-        // Give priority to failed event from reactor over MessagingException event.
-        BaseEventContext context = (BaseEventContext) (event != null ? event.getContext()
-            : ((MessagingException) throwable).getEvent().getContext());
-        errorNotification(processor).andThen(e -> context.error(e))
-            .accept(resolveMessagingException(processor).apply((MessagingException) throwable));
-      } else {
-        if (event == null) {
+        throwable = Exceptions.unwrap(throwable);
+
+        if (throwable instanceof MessagingException) {
+          BaseEventContext context = (BaseEventContext) ((MessagingException) throwable).getEvent().getContext();
+          errorNotification(processor).andThen(e -> context.error(e))
+              .accept(resolveMessagingException(processor).apply((MessagingException) throwable));
+        } else {
           LOGGER.error(UNEXPECTED_ERROR_HANDLER_STATE_MESSAGE, throwable);
           throw new IllegalStateException(UNEXPECTED_ERROR_HANDLER_STATE_MESSAGE);
-        } else {
-          BaseEventContext context = ((BaseEventContext) event.getContext());
+        }
+      } else {
+        CoreEvent event = (CoreEvent) object;
+        throwable = Exceptions.unwrap(throwable);
+        if (throwable instanceof MessagingException) {
+          // Give priority to failed event from reactor over MessagingException event.
+          BaseEventContext context = (BaseEventContext) (event != null ? event.getContext()
+              : ((MessagingException) throwable).getEvent().getContext());
           errorNotification(processor).andThen(e -> context.error(e))
-              .accept(resolveException(processor, event, throwable));
+              .accept(resolveMessagingException(processor).apply((MessagingException) throwable));
+        } else {
+          if (event == null) {
+            LOGGER.error(UNEXPECTED_ERROR_HANDLER_STATE_MESSAGE, throwable);
+            throw new IllegalStateException(UNEXPECTED_ERROR_HANDLER_STATE_MESSAGE);
+          } else {
+            BaseEventContext context = ((BaseEventContext) event.getContext());
+            errorNotification(processor).andThen(e -> context.error(e))
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7683&issues=AWjyMA3aotIXnmsUSXGj&open=AWjyMA3aotIXnmsUSXGj)

[ISSUE_KEY]: AWjyMA3aotIXnmsUSXGj